plugins/services/src/js/stores/SDKEndpointStore.js

```diff
@@ -0,0 +1,154 @@
+  constructor() {
+    super(...arguments);
+
+    const self = this;
+
+    this.getSet_data = {
+      services: {}
+    };
+
+    PluginSDK.addStoreConfig({
+      store: this,
+      storeID: this.storeID,
+      listenAlways: true
+    });
+
+    this.dispatcherIndex = AppDispatcher.register(function(payload) {
+      if (payload.source !== SERVER_ACTION) {
+        return false;
+      }
+
+      const { type, data } = payload.action;
+      switch (type) {
+        case REQUEST_SDK_ENDPOINTS_SUCCESS:
+          self.processEndpoints(data.serviceId, data.endpoints);
+          break;
+        case REQUEST_SDK_ENDPOINTS_ERROR:
+          self.setService(data.serviceId, {
+            endpoints: [],
+            error: data.error.description
+          });
+          break;
+        case REQUEST_SDK_ENDPOINT_SUCCESS:
+          self.processEndpoint(
+            data.serviceId,
+            data.endpointName,
+            data.endpointData,
+            data.contentType
+          );
+          break;
+        case REQUEST_SDK_ENDPOINT_ERROR:
+          self.setService(data.serviceId, {
+            endpoints: [],
+            error: data.error
+          });
+          break;
+      }
+
+      return true;
+    });
+  }
+
+  getServices() {
+    return Util.deepCopy(this.get("services"));
+  }
+
+  getServiceEndpoints(serviceId) {
+    const service = this.getServices()[serviceId];
+
+    if (!service || !service.endpoints) {
+      return null;
+    }
+
+    return Object.entries(service.endpoints).map(
+      ([endpointName, endpoint]) =>
+        new ServiceEndpoint({
+          endpointName,
+          endpointData: endpoint.endpointData,
+          contentType: endpoint.contentType
+        })
+    );
+  }
+
+  getServiceError(serviceId) {
+    const service = this.getServices()[serviceId];
+
+    if (!service || !service.error) {
+      return "";
+    }
+
+    return service.error;
+  }
+
+  setService(serviceId, serviceData) {
+    const services = this.getServices();
+    services[serviceId] = {
+      endpoints: serviceData.endpoints,
+      error: serviceData.error
+    };
+
+    this.set({ services });
+  }
+
+  processEndpoints(serviceId, endpointsArray) {
+    const endpoints = endpointsArray.reduce(function(acc, endpointName) {
+      acc[endpointName] = {};
+
+      return acc;
+    }, {});
+
+    this.setService(serviceId, {
+      endpoints,
+      error: ""
+    });
+
+    endpointsArray.forEach(function(endpoint) {
+      SDKEndpointActions.fetchEndpoint(serviceId, endpoint);
+    }, this);
+  }
+
+  processEndpoint(serviceId, endpointName, endpointData, contentType) {
+    const service = this.getServices()[serviceId];
+    if (!service || !service.endpoints[endpointName]) {
+      return;
+    }
+    service.endpoints[endpointName] = { endpointData, contentType };
+
+    this.setService(serviceId, {
+      endpoints: service.endpoints,
+      error: ""
+    });
+  }
+
+  addChangeListener(eventName, callback) {
```

Comment:
We inherit`addChangeListener` and  `removeChangeListener ` through the prototype chain so no need to declare again.